File: app/src/main/java/com/absinthe/kage/device/Command.java
package com.absinthe.kage.device;

import com.absinthe.kage.connect.protocol.IpMessageProtocol;
import com.absinthe.kage.device.client.Client;

public abstract class Command {

    protected static final String DELIMITER = IpMessageProtocol.DELIMITER;
    protected int cmd;

    protected Command() {}
    public abstract String pack();
    public abstract void doWork(Client client, String received);
    public abstract boolean parseReceived(String received);
}

File: app/src/main/kotlin/com/absinthe/kage/device/Command.kt
package com.absinthe.kage.device

import com.absinthe.kage.connect.protocol.IpMessageProtocol
import com.absinthe.kage.device.client.Client

abstract class Command protected constructor() {

    var cmd = 0

    abstract fun pack(): String
    abstract fun doWork(client: Client, received: String)
    abstract fun parseReceived(received: String): Boolean

    companion object {
        const val DELIMITER = IpMessageProtocol.DELIMITER
    }
}